torch/nn/functional.py

```diff
@@ -3829,26 +3829,125 @@ def fold(input, output_size, kernel_size, dilation=1, padding=0, stride=1):
 def _pad_circular(input, padding):
     # type: (Tensor, List[int]) -> Tensor
     """
-    Arguments
-        :param input: tensor of shape :math:`(N, C_{\text{in}}, H, [W, D]))`
-        :param padding: (tuple): m-elem tuple where m is the degree of convolution
-    Returns
-        :return: tensor of shape :math:`(N, C_{\text{in}}, [D + 2 * padding[0],
-                 H + 2 * padding[1]], W + 2 * padding[2]))`
+    Args:
+        input: Tensor that follows the formatting of the input to convolution
+            layers.
+        padding: Tuple with length two times the degree of the convolution. The
+            order of the integers in the tuple are shown in the following
+            example:
+
+            For 3D convolutions:
+                padding[-2] is the amount of padding applied to the beginning
+                    of the depth dimension.
+                padding[-1] is the amount of padding applied to the end of the
+                    depth dimension.
+                padding[-4] is the amount of padding applied to the beginning
+                    of the height dimension.
+                padding[-3] is the amount of padding applied to the end of the
+                    height dimension.
+                padding[-6] is the amount of padding applied to the beginning
+                    of the width dimension.
+                padding[-5] is the amount of padding applied to the end of the
+                    width dimension.
+
+    Returns:
+        out: Tensor with padded shape.
     """
+    shape = input.shape
+    ndim = len(shape[2:])
```

Comment:
nit: `shape[2:]` is going to be used several times, maybe give it a name?

Reply:
Done!